samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -150,6 +152,13 @@
    */
   private final Optional<StreamRegexMonitor> inputStreamRegexMonitor;
 
+  /**
+   * Container placement action dispatcher and util
+   */
+  private final ContainerPlacementUtil containerPlacementUtil;
```

Comment:
Either rename this to ContainerPlacementMetastoreReader or make all its methods static so that it is really a Util. 
A Util should not be instantiated like this and have a complete lifecycle.

Reply:
+1